source/Octopus.Tentacle.Client/TentacleClient.cs

```diff
@@ -209,6 +211,136 @@ async Task<DataStream> DownloadFileAction(CancellationToken ct)
             }
         }
 
+
+        public async Task<ScriptStatusResponseV3Alpha> StartScript(
```

Comment:
Would it be possible to return something that is not the type that is received over the wire.

Perhaps two things distinct objects should be returned here
 - The interesting data returned e.g. logs.
 - An object which can be passed to `GetStatus` containing everything required to do the next `GetStatus` call. This object could additionally have some extension methods for serialization and deserialization, which is all taken care of in the client (and tested in the client).

Reply:
Then we could use `CommandContext` as the parameter to the other commands, to really highlight this is what you need to use these.

Reply:
nice, in the future it would be nice if the client can tell us if the script is hung (which is after the script has started and it has not logged output for more than some amount of time). Could we stash the information needed to do that in the command context?

Reply:
Yeah, probably. 

This sounds like a nice separate piece of work to come afterwards. I'll create a card for it.

Reply:
Done.

Also, [story created](https://app.shortcut.com/octopusdeploy/story/71068/detect-if-tentacle-client-has-hung)